client/components/domains/registrant-extra-info/disable-submit-button.jsx

```diff
@@ -0,0 +1,15 @@
+import { isEmpty, map } from 'lodash';
+import { cloneElement } from 'react';
+
+export function disableSubmitButton( children ) {
+	if ( isEmpty( children ) ) {
+		return children;
+	}
+
+	return map( Array.isArray( children ) ? children : [ children ], ( child, index ) =>
```

Comment:
This function should use `React.Children.map` but it seems it doesn't matter because ... the function is never called with non-empty children and could be removed, too?

That's because the `<RegistrantExtraInfoForm />` component is never rendered with children. Can you verify that I followed the call path correctly? From `RegistrantExtraInfoForm` to the `{ca,fr,uk}`-specific forms in `DomainDetailsForm` and to the `{ca,fr,uk}-form.js` modules that call `disableSubmitButton` on the always-empty `children`?

Reply:
Great catch 😉  

I've traced the potential children to the same path you described, and yes, `RegistrantExtraInfoForm` never renders with children. I'll clean this up 👍

Reply:
Cleaned up in https://github.com/Automattic/wp-calypso/pull/58396/commits/d05621e90d55fb9bcba9bcc6a462289311853727. Then removed some more unused related code in https://github.com/Automattic/wp-calypso/pull/58396/commits/928c77837efbbcda74236f41e64b6aa8ae821733 and some irrelevant children tests in https://github.com/Automattic/wp-calypso/pull/58396/commits/b9e17c3863c1ebdbdeaf0120fff75a0f35e9f00b. This is ready for another look 👀